compiler_opt/rl/feature_ops.py

```diff
@@ -80,3 +85,88 @@ def normalize(obs: types.Float):
     return tf.concat(features, axis=-1)
 
   return normalize
+
+
+def get_ir2vec_normalize_fn(ir2vec_with_z_score_normalization: bool = False,
+                            eps: float = 1e-8):
+  """Return a normalization function for embeddings.
+
+  Args:
+    ir2vec_with_z_score_normalization: If True, applies z-score standardization
+      (whitening) to the embeddings per batch. This transforms the embeddings
+      to have zero mean and unit variance across the batch dimension.
+      Setting this to True would reduce the impact of outlier embeddings and
+      improve training stability, especially when embeddings from different
+      batches have significantly different scales.
+
+    eps: Small epsilon value added to standard deviation to prevent division
+      by zero during standardization.
+
+  Returns:
+    A function that normalizes IR2Vec embeddings according to the specified
+    parameters.
+  """
+  if ir2vec_with_z_score_normalization:
+    # Whitens the embeddings per batch.
+    def z_score_normalize(batch_embeddings: types.Float):
+      mean = tf.math.reduce_mean(batch_embeddings, axis=0, keepdims=True)
+      std = tf.math.reduce_std(batch_embeddings, axis=0, keepdims=True)
+      standardized_embeddings = (batch_embeddings - mean) / (std + eps)
+      return standardized_embeddings
+
+    return z_score_normalize
+  # Currently, we just return the identity function for embeddings.
+  # We can extend this to include other normalizations like L2
+  # normalization if needed.
+  return lambda x: identity_fn(x, expand_dims=False)
```

Comment:
Why does this need to be `identity_fn` here instead of just `tf.cast` (maybe bound with `functools.partial`)?